containers/src/impl/Kokkos_UnorderedMap_impl.hpp

```diff
@@ -76,19 +75,19 @@ struct UnorderedMapErase {
   using size_type       = typename map_type::size_type;
   using key_type        = typename map_type::key_type;
   using value_type      = typename map_type::impl_value_type;
+  using policy_type =
+      Kokkos::RangePolicy<execution_space, IndexType<size_type>>;
 
   map_type m_map;
 
-  UnorderedMapErase(map_type const& map) : m_map(map) {}
```

Comment:
Why did you remove?

Reply:
1. `UnorderedMapErase` is an aggregate and as such can be initialized using aggregate initialization.
2. In my opinion, the constructor was not correct in the sense that, bearing in mind that the map is a "shared pointer to some memory", it should have been `UnorderedMapErase(map_type map) : m_map(std::move(map)) {}` to clearly show that shared ownership is requested.

All in all I decided to remove the constructor :pencil2:

Reply:
@masterleinad What's your opinion on this then ? :)

Reply:
My preference is to minimize drive-by changes. 🙂

Reply:
@masterleinad I already extracted these changes from #6584... I'm not against making another "extracted from" PR with the "remove useless constructors" change if you really want to and in general I agree with the "one PR per topic without drive-by change", but it makes the contribution process very lengthy for such a "deletion of useless line of code change" case... Especially that in this PR I already modify the `UnorderedMapErase` & Co. functors, so deleting their useless constructor can arguably not be a drive-by change :angel:

What do you think?

Reply:
You likely get a pull request merged if you change fewer lines of code but it's not a showstopper for me.